browser/ui/webui/brave_new_tab/new_tab_page_handler.h

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_UI_WEBUI_BRAVE_NEW_TAB_NEW_TAB_PAGE_HANDLER_H_
+#define BRAVE_BROWSER_UI_WEBUI_BRAVE_NEW_TAB_NEW_TAB_PAGE_HANDLER_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/memory/raw_ptr.h"
+#include "brave/browser/ui/webui/brave_new_tab/update_observer.h"
+#include "brave/components/brave_new_tab/common/new_tab_page.mojom.h"
+#include "mojo/public/cpp/bindings/pending_receiver.h"
+#include "mojo/public/cpp/bindings/pending_remote.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+#include "mojo/public/cpp/bindings/remote.h"
+
+class CustomBackgroundFileManager;
+class PrefService;
+
+namespace ntp_background_images {
+class ViewCounterService;
+}
+
+namespace brave_new_tab {
+
+class CustomImageChooser;
+
+class NewTabPageHandler : public mojom::NewTabPageHandler {
+ public:
+  NewTabPageHandler(
+      mojo::PendingReceiver<mojom::NewTabPageHandler> receiver,
+      std::unique_ptr<CustomImageChooser> custom_image_chooser,
+      std::unique_ptr<CustomBackgroundFileManager> custom_file_manager,
+      PrefService* pref_service,
+      ntp_background_images::ViewCounterService* view_counter_service,
+      bool is_restored_page);
```

Comment:
why do we need to know if it's restored or not?

Reply:
This came from the current NTP. It may be a Chesterton’s Fence situation, but I'll remove it for now.

Reply:
Fixed.